src/GafferOSL/ShadingEngine.cpp

```diff
@@ -294,6 +294,22 @@ class RenderState
 				t.arraylen = 2;
 				return ShadingSystem::convert_value( value, type, src, t );
 			}
+			else if ( it->dataView.type.basetype == TypeDesc::DOUBLE )
+			{
+				double doubleValue;
+				ShadingSystem::convert_value( &doubleValue, TypeDesc::DOUBLE, src, TypeDesc::DOUBLE );
```

Comment:
This is fairly performance-critical code, so I think it's probably worthwhile preferring `static_cast<const double *>( src )` over `convert_value()` here.

Reply:
favoured casts over function call in ba245a1